memcache/memcache.go

```diff
@@ -519,26 +511,15 @@ func (c *Client) parseGetResponse(r *bufio.Reader, cb func(*Item)) error {
 			return err
 		}
 		buffSize := size + 2
-		if c.Pool != nil {
-			v, err := c.Pool.Get(buffSize)
-			if err != nil {
-				return err
-			}
-			it.Value = (*v)[:buffSize]
-		} else {
-			it.Value = make([]byte, buffSize)
-		}
+		buff := opts.Alloc.Get(buffSize)
+		it.Value = *buff
```

Comment:
I've elected to _not_ force the length of the buffer like the previous code did. This is because the interface for `Allocator` guarantees that the returned buffer will be correctly sized.

Reply:
Looking at the old `BytesPool` interface, it didn't guarantee length, only capacity. WDYT about taking that approach here? Seems a bit less error prone and moves some of the burden onto this code to assert the correct length vs making `Allocator` implementations do it.